st2api/st2api/controllers/v1/actionviews.py

```diff
@@ -98,32 +95,36 @@ class OverviewController(resource.ContentPackResourceController):
 
     include_reference = True
 
-    @jsexpose(arg_types=[str])
     def get_one(self, ref_or_id):
         """
             List action by id.
 
             Handle:
                 GET /actions/views/overview/1
         """
-        action_api = super(OverviewController, self)._get_one(ref_or_id)
-        return self._transform_action_api(action_api)
+        resp = super(OverviewController, self)._get_one(ref_or_id)
+        action_api = resp.json
+        result = self._transform_action_api(action_api)
+        resp.json = result
+        return resp
 
-    @jsexpose(arg_types=[str])
     def get_all(self, **kwargs):
         """
             List all actions.
 
             Handles requests:
                 GET /actions/views/overview
         """
-        action_apis = super(OverviewController, self)._get_all(**kwargs)
-        return map(self._transform_action_api, action_apis)
+        resp = super(OverviewController, self)._get_all(**kwargs)
+        action_apis = resp.json
+        result = map(self._transform_action_api, action_apis)
+        resp.json = result
+        return resp
 
     @staticmethod
     def _transform_action_api(action_api):
-        action_id = action_api.id
-        action_api.parameters = ParametersViewController._get_one(action_id).get('parameters')
+        action_id = action_api['id']
```

Comment:
Not that I had a moment to check the code you're commenting, this particular place have an asterisk to it, really. We accessing it as a dict and not an object because at that point we're already working not with user input, but with server response. Method `resp = super(OverviewController, self)._get_all(**kwargs)` already created a response for us and now we're trying to modify it.

Now, if you want to work with the model here, `._get_all` method should also return model, which is possible, yet would require us to make sure `.get_all` of every other controller in the whole st2api will convert it to Response at the end.

Although we still support controllers returning models, I find this practice highly detrimental to us better understanding how http stack works. Which was part of the reason I'm doing it to begin with. Pecan was complex and involved too many features we don't need and don't care. Router is short enough anyone on the team can read through it and understand what's going on in between of user making a request and receiving the answer to it. I consider understanding of the Router.__call__ method an absolute requirement for anyone touching an endpoint on this project.